xla/service/latency_hiding_scheduler.cc

```diff
@@ -1901,10 +1907,23 @@ absl::StatusOr<HloGraphNode::TimeCost> DefaultSchedulerCore::ScheduleNode(
   }
   ++sched_state->scheduled_count;
   for (auto& resource : n->GetResources()) {
-    if (resource.second == ResourceUsageType::kResourceRelease) {
-      --sched_state->resources_in_flight[resource.first];
+    // No need to release non-extendable resources
+    if (resource.second == ResourceUsageType::kResourceRelease &&
```

Comment:
@seherellis is this if-condition sound? i.e. can we not release non extendable resources here? I think it was introduced here to fix one of the LHS tests, but I'm not sure if it's the proper fix.

@terryysun to potentially five more context.

Reply:
Good catch! This is not a proper fix. 

@terryysun why did you need to add the following in this condition?
`async_tracker_->GetResourceHazardType(resource.first) != ResourceHazardType::kNonextendable`